gobblin-data-management/src/main/java/gobblin/data/management/policy/NewestKSelectionPolicy.java

```diff
@@ -28,30 +29,89 @@
 /**
  * Select the newest k versions of the dataset.
  */
-public class NewestKSelectionPolicy<T extends DatasetVersion> implements VersionSelectionPolicy<T> {
+public class NewestKSelectionPolicy implements VersionSelectionPolicy<DatasetVersion> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(NewestKSelectionPolicy.class);
 
+  /**
+   * The number of newest versions to select. Only one of
+   * {@link #NEWEST_K_VERSIONS_SELECTED_KEY} and {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} can
+   * be specified. The default is {@link #NEWEST_K_VERSIONS_SELECTED_KEY} with a value of
+   * {@link #VERSIONS_SELECTED_DEFAULT_INT}. Valid values are in the range
+   * [{@link #MIN_VERSIONS_ALLOWED}, {@link #MAX_VERSIONS_ALLOWED}]. Negative values are interpreted
+   * as {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} with the absolute value.
+   */
   public static final String NEWEST_K_VERSIONS_SELECTED_KEY = "selection.newestK.versionsSelected";
 
-  public static final String VERSIONS_SELECTED_DEFAULT = Integer.toString(2);
+  /**
+   * The number of newest versions to exclude from the result. Only one of
+   * {@link #NEWEST_K_VERSIONS_SELECTED_KEY} and {@link #NEWEST_K_VERSIONS_NOTSELECTED_KEY} can
+   * be specified. The default is {@link #NEWEST_K_VERSIONS_SELECTED_KEY} with a value of
+   * {@link #VERSIONS_SELECTED_DEFAULT}. Valid values are in the range
+   * [{@link #MIN_VERSIONS_ALLOWED}, {@link #MAX_VERSIONS_ALLOWED}]. Negative values are interpreted
+   * as {@link #NEWEST_K_VERSIONS_SELECTED_KEY} with the absolute value.
+   */
+  public static final String NEWEST_K_VERSIONS_NOTSELECTED_KEY = "selection.newestK.versionsNotSelected";
 
+  public static final Integer VERSIONS_SELECTED_DEFAULT = 2;
+
+  public static final Integer MAX_VERSIONS_ALLOWED = 1000000;
+  public static final Integer MIN_VERSIONS_ALLOWED = - MAX_VERSIONS_ALLOWED;
+
+  /** Number of versions to keep if non-negative; number of versions to exclude if negative. */
   private final int versionsSelected;
 
   public NewestKSelectionPolicy(int versionsRetained) {
     this.versionsSelected = versionsRetained;
-    LOGGER.info(String.format("%s will select %d versions of each dataset.", NewestKSelectionPolicy.class.getName(),
+    LOGGER.info(String.format("Will %s %d versions of each dataset.",
+        (this.versionsSelected >= 0 ? "select" : "exclude"),
         this.versionsSelected));
   }
 
   public NewestKSelectionPolicy(Properties props) {
-    this.versionsSelected =
-        Integer.parseInt(props.getProperty(NEWEST_K_VERSIONS_SELECTED_KEY, VERSIONS_SELECTED_DEFAULT));
+    this(getNumVersionsFromProps(props));
```

Comment:
I don't think so. This is a fairly new class. It should be fine to parse props as config.